compliance/cis_k8s/rules/cis_4_2_2/rule.rego

```diff
@@ -6,20 +6,29 @@ import data.compliance.lib.common
 import data.compliance.lib.data_adapter
 
 # Ensure that the --authorization-mode argument is not set to AlwaysAllow (Automated)
-
 # If the --authorization-mode argument is present check that it is not set to AlwaysAllow.
 
+default rule_evaluation = false
+
 # evaluate
 process_args := data_adapter.process_args
 
-default rule_evaluation = false
-
 rule_evaluation {
+	is_authorization_allow_all
+}
+
+is_authorization_allow_all {
 	process_args["--authorization-mode"]
-	assert.is_false(common.arg_values_contains(process_args, "--authorization-mode", "AlwaysAllow"))
+	not common.contains_key_with_value(process_args, "--authorization-mode", "AlwaysAllow")
```

Comment:
I think we should revert this change since it can contain more than one value, i.e.
```rego
rule_input("--authorization-mode=AlwaysAllow,RBAC")
```
Also, can you add tests with more than one authorization mode, please?